### ISSUE TITLE
Expanding features and updating

### ISSUE THREAD

**Issue by davorpeic** [progress/estimote-nativeScript-plugin#7]
Hi,

are there any plans to expand to more features? We are interested in monitoring and indoor location? We could also contribute financially if that would help to get it on

thanks

**Comment by ghost** [comment]
+1

**Comment by davorpeic** [comment]
Small update for this repo, I managed to update the necessary files/libraries to get this working in Nativescript 2.5. My fork is based on albertocule's fork where he fixed Android version, and in my version I fixed iOS build and provided example on how to use in angular+typescript project, so now both are building fine. I tested it on Android 4.4.2 phone and iphone 4s (ios 8).

At this moment I will keep that fork live and will send PR to this repo, but I can't guarantee it will be merged..

https://github.com/davorpeic/estimote-nativeScript-plugin
I recommend cloning the repo to local, and then run install from there, for some reason can't install it directly from github.
`tns plugin add "path-to-this-folder/estimote-nativeScript-plugin/Plugin"`

Roadmap:
- Remove EstimoteSDK from the repo (lower the repo size for 200+ MB) and use Cocoapods _(already done locally and in process of testing)_
- Update SDK to latest versions (Both Android & iOS)
- Update plugin to use typescript by default.
- Add more methods that are available from SDK

Not sure yet, but I will most probably create new plugin as the majority of the code will be new, or not sure if Progress is willing to pass this repo ownership to us, will decide in next few weeks.

Also, I'm pretty new to nativescript plugin development so any contributions are more then welcomed :)

dp

**Comment by albertocule** [comment]
Good job @davorpeic !

**Comment by irums** [comment]
Unable to get this plugin to work on NativeScript 2.5.2.

**Comment by ghost** [comment]
I can confirm that it works with 2.5.3

**Comment by ghost** [comment]
It seems that the plugin from @davorpeic doesn't work when there are any spaces in the path of the project. If I copy the project to a spaceless folder, it works like a charm (thanks for that!!).

```
Processing node_modules failed. Error: Command failed: /bin/sh -c lipo -i /Users/andy/Google Drive/Aktuelle Projekte & Kunden/Josera GmbH & Co. KG/Apps/JOSERA Sales Support/Source 1.0/JOSERASalesSupport/node_modules/nativescript-estimote-sdk/platforms/ios/libEstimoteSDK.a
/bin/sh: Kunden/Josera: No such file or directory
/bin/sh: Co.: command not found
fatal error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: can't open input file: /Users/andy/Google (No such file or directory)
```

**Comment by davorpeic** [comment]
Hey @spacetasse , I normally avoid using dropbox/gdrive for projects as I don't like to be syncing all files that are not necessarily needed (node staff) and I found many permissions issues in the past.

Is this the only nativescript app that fails or other apps in gdrive also fail to build? Algo can you check if that folder/file exist on that path just to be sure npm installed the libraries (possible permission issue)

thanks

**Comment by ghost** [comment]
Hey Davor,

any other project in gdrive works and before I add the plugin, this project runs also. After I add the plugin, I get that error. If I copy the directory to an „non-spaced“-location, everything works fine.

I had the same issue with an other project, too. 

Thanks :)
Andy

**Comment by irums** [comment]
HI davorpeic

It works on android, but it does not work with ios. do i need to do anything about permissions for it to be able to access the bluetooth features of the phone?

**Comment by davorpeic** [comment]
Hm, 

let me try to update my repo to latest {NS} and see what could be the issue, please give a day or two

**Comment by irums** [comment]
ok. thanks

**Comment by irums** [comment]
It doesnt work on ios because u use this in ios 
global.ESTIMOTE_PROXIMITY_UUID = NSUUID.alloc().initWithUUIDString("85FC11DD-4CCA-4B27-AFB3-876854BB5C3B");

and u use this in android.

global.ESTIMOTE_PROXIMITY_UUID = java.util.UUID.fromString("B9407F30-F5F8-466E-AFF9-25556B57FE6D");


the android UUID is the one that works. i changed it in the ios and both work.

**Comment by davorpeic** [comment]
@irums yeah, you should put there your UUIDs, but it is strange that you don't define those params when calling plugin, I will see if I can do something about it, I thought it had those options defined outside, its been a while since I played with this plugin.

**Comment by ghost** [comment]
I'm using this plugin within several projects, but on one project I get.
`file:///app/tns_modules/nativescript-estimote-sdk/estimote.js:65:38: JS ERROR ReferenceError: Can't find variable: region`

The error occurs only on iOS. The Android version runs smoothly. Region is defined in the options.

**Comment by ghost** [comment]
URGENT: It seems that the ranging don't work on some Samsung and Motorola-Phones. In our test environment runs all smoothly, but in real live, some devices even see beacons :-( That is currently a BIG problem for us :-(

**Comment by deividfortuna** [comment]
Is there any plan to fix things or progress give up of this project?
